examples/fit_ODNP_data.py

```diff
@@ -0,0 +1,119 @@
+"""Fitting ODNP Datasets for Ksigma
+===================================
+The T1(p) and E(p) integrals are stored in previous post processing pulled from an H5 file. The inverse of the T1(p) data is taken to produce the R1(p) data which is fit using the polyfit of the inverse of krho with two degrees of freedom. The R1(p) fit is then fed into the fitting routine for the enhancement data prior to normalization. The cross relaxivity is then calculated using the output of these fits as well as the sample parameters that are fed to it (e.g., ppt value, and concentration).
+"""
+from pyspecdata import *
+from sympy import symbols, Symbol, latex
+from scipy.io import loadmat
+
+target_directory = os.path.normpath(getDATADIR("AG_processed_data"))
+h5_file = "ras.h5"
+nodename = "230706_M67_a"
+# {{{ plotting fn - for labels on final plot
+def list_symbs(f):
+    list_symbs = [f"${latex(Symbol(j))} = {k:0.5g}$" for j, k in f.output().items()]
+    list_symbs = "\n".join(list_symbs)
+    return list_symbs
+
+
+# }}}
+# {{{ load data
+Ep = nddata_hdf5(f"{h5_file}/{nodename}/Ep", directory=target_directory)
+R1p = nddata_hdf5(f"{h5_file}/{nodename}/R1p", directory=target_directory)
+# }}}
+# {{{Dataset Parameters
+SL_conc_M = Ep.get_prop("acq_params")["concentration"]
+ppt = Ep.get_prop("acq_params")["guessed_MHz_to_GHz"] * 1e-3
+Ep_pts = Ep.getaxis("power")
+# {{{Find the index where the return powers begin
+thisdiff = Ep.fromaxis("power").diff("power")
+for idx in range(len(Ep.getaxis("power"))):
+    if thisdiff["power", idx].data < 0:
+        flip_idx = idx
+        break
+flip_idx += 1
+# }}}
+# }}}
+with figlist_var() as fl:
+    # {{{Plot Ep
+    fl.next("Integrated Enhancement")
+    fl.plot(
+        Ep["power", :flip_idx],
+        "o",
+        label="Data for increasing power",
+        capsize=6,
+        alpha=0.5,
+    )
+    fl.plot(
+        Ep["power", flip_idx:],
+        "rx",
+        label="Returning Power Check",
+        capsize=6,
+        alpha=0.5,
+    )
+    # }}}
+    # {{{ plot R1p data
+    fl.next(r"$R_{1}(p)$")
+    fl.plot(R1p, "o", label="Experimental Data")
+    # }}}
+    # {{{load in T100 dataset
+    T10_p = loadmat("T10_DI_water_230412")["a"][0, :]
+    R10_p = nddata((T10_p[0] + T10_p[1] * R1p.getaxis("power")) ** -1, "power")
```

Comment:
Use 'fromaxis'

Reply:
`Traceback (most recent call last):
  File "fit_ODNP_data.py", line 61, in <module>
    R10_p = nddata((T10_p[0] + T10_p[1] * R1p.fromaxis("power")) ** -1, "power")
  File "c:\users\alex guinness\gitstuff\pyspecdata\pyspecdata\core.py", line 2170, in __init__
    if len(args[0].shape) == 1 and isinstance(args[1], str):
  File "c:\users\alex guinness\gitstuff\pyspecdata\pyspecdata\core.py", line 5666, in __getattribute__
    return super().__getattribute__(arg)
`